python/pyarrow/tests/test_parquet.py

```diff
@@ -417,26 +470,29 @@ def test_pandas_parquet_1_0_roundtrip(tempdir):
 
 
 @pytest.mark.pandas
-def test_multiple_path_types(tempdir):
+@parametrize_use_dataset
+def test_multiple_path_types(tempdir, use_dataset):
     # Test compatibility with PEP 519 path-like objects
     path = tempdir / 'zzz.parquet'
     df = pd.DataFrame({'x': np.arange(10, dtype=np.int64)})
     _write_table(df, path)
-    table_read = _read_table(path)
+    table_read = _read_table(path, use_dataset=use_dataset)
     df_read = table_read.to_pandas()
     tm.assert_frame_equal(df, df_read)
 
     # Test compatibility with plain string paths
     path = str(tempdir) + 'zzz.parquet'
     df = pd.DataFrame({'x': np.arange(10, dtype=np.int64)})
     _write_table(df, path)
-    table_read = _read_table(path)
+    table_read = _read_table(path, use_dataset=use_dataset)
     df_read = table_read.to_pandas()
     tm.assert_frame_equal(df, df_read)
 
 
+# TODO(dataset) duplicate column selection actually gives duplicate columns now
```

Comment:
This seems like it could be fixed just for ParquetDatasetV2 by uniqueing the column names, is that unfavorable?

Reply:
Yes, that should be easy to do. 
But, I would prefer that we decide on this for the new Datasets API what we want (deduplicating the passed columns, or returning duplicated columns), and follow that here. Otherwise it creates an inconsistency between this and the `pyarrow.dataset` API. So therefore I left it as a TODO for now (with the TODO being the need to bring this up / take a decision).

Reply:
I think we will not deduplicate the column names in C++ @fsaintjacques 

Fields within a schema may have duplicated field names so it seems unlikely that we'll move amenities like deduplication to the lowest level.